fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java

```diff
@@ -152,17 +152,26 @@ public EsTable(long id, String name, List<Column> schema, TableType tableType) {
     }
 
     public Map<String, String> fieldsContext() throws UserException {
+        makeSureEsMetaSynced();
         return esMetaStateTracker.searchContext().fetchFieldsContext();
     }
 
     public Map<String, String> docValueContext() throws UserException {
+        makeSureEsMetaSynced();
         return esMetaStateTracker.searchContext().docValueFieldsContext();
     }
 
     public List<String> needCompatDateFields() throws UserException {
+        makeSureEsMetaSynced();
         return esMetaStateTracker.searchContext().needCompatDateFields();
     }
 
+    private void makeSureEsMetaSynced() {
+        if (esMetaStateTracker == null) {
```

Comment:
Do we need to consider concurrency issue?

Reply:
No need here. We only ensure that meta tracker is not null before the master demon process finished  and this method only be used in ES external table not ES catalog.